packages/api/lib/EarthdataLogin.js

```diff
@@ -123,11 +130,50 @@ class EarthdataLogin extends OAuth2 {
     try {
       const response = await this.requestAccessToken(authorizationCode);
 
+      log.debug(`expires in: ${response.expires_in}`);
+
+      return {
+        accessToken: response.body.access_token,
+        refreshToken: response.body.refresh_token,
+        username: response.body.endpoint.split('/').pop(),
+        expirationTime: Date.now() + this.getTokenExpirationMs()
```

Comment:
From @flamingbear: Why are we not using the expiration provided by the OAuth provider directly?

Reply:
I looked into this and I can't think of a reason why we shouldn't use the `response.expires_in` from the OAuth provider. For Earthdata the value is returned in seconds and seems to be `3600`, so we can set the `expirationTime` to the current time + `response.expires_in` and it will expire in an hour.